pkg/datastore/datastore.go

```diff
@@ -175,7 +176,7 @@ func (ds *memoryDatastore) Write(opsrc *marketplace.OperatorSource, registryMeta
 	return
 }
 
-func (ds *memoryDatastore) GetRepositoryByPackageName(pkg string) (repository *Repository, err error) {
+func (ds *memoryDatastore) GetRepositoryByOperatorSourceAndPackageName(sourceName, pkg string) (repository *Repository, err error) {
```

Comment:
It seems to me that, though this is a smaller code change, having variables that can be set to nil and result in different behavior could be difficult to read down the line. Maybe at the Datastore level these should be two separate functions?

Also this method name is really long.

Reply:
I suggest using `GetRepository`